go/vt/schemadiff/schema.go

```diff
@@ -282,3 +285,150 @@ func (s *Schema) Diff(other *Schema, hints *DiffHints) (diffs []EntityDiff, err
+func (s *Schema) ToQueries() []string {
+	queries := []string{}
+	for _, e := range s.Entities() {
+		queries = append(queries, e.Create().CanonicalStatementString())
+	}
+	return queries
+}
+
+// ToSQL returns a SQL blob with ordered sequence of queries which can be applied to create the schema
+func (s *Schema) ToSQL() string {
+	var buf bytes.Buffer
+	for _, query := range s.ToQueries() {
+		buf.WriteString(query)
+		buf.WriteString(";\n")
+	}
+	return buf.String()
+}
+
+// apply attempts to apply given list of diffs to this object.
+// These diffs are CREATE/DROP/ALTER TABLE/VIEW.
+func (s *Schema) apply(diffs []EntityDiff) error {
+	for _, diff := range diffs {
+		switch diff := diff.(type) {
+		case *CreateTableEntityDiff:
+			// We expect the table to not exist
+			if _, ok := s.named[diff.createTable.Table.Name.String()]; ok {
+				return ErrApplyDuplicateTableOrView
+			}
+			s.tables = append(s.tables, &CreateTableEntity{CreateTable: *diff.createTable})
+		case *CreateViewEntityDiff:
+			// We expect the view to not exist
+			if _, ok := s.named[diff.createView.ViewName.Name.String()]; ok {
+				return ErrApplyDuplicateTableOrView
+			}
+			s.views = append(s.views, &CreateViewEntity{CreateView: *diff.createView})
+		case *DropTableEntityDiff:
+			// We expect the table to exist
+			found := false
+			for i, t := range s.tables {
+				if t.Table.Name.String() == diff.from.Table.Name.String() {
+					s.tables = append(s.tables[0:i], s.tables[i+1:]...)
+					delete(s.named, t.Table.Name.String())
+					found = true
+					break
+				}
+			}
+			if !found {
+				return ErrApplyTableNotFound
+			}
+		case *DropViewEntityDiff:
+			// We expect the view to exist
+			found := false
+			for i, v := range s.views {
+				if v.ViewName.Name.String() == diff.from.ViewName.Name.String() {
+					s.views = append(s.views[0:i], s.views[i+1:]...)
+					delete(s.named, v.ViewName.Name.String())
+					found = true
+					break
+				}
+			}
+			if !found {
+				return ErrApplyViewNotFound
+			}
+		case *AlterTableEntityDiff:
+			// We expect the table to exist
+			found := false
+			for i, t := range s.tables {
+				if t.Table.Name.String() == diff.from.Table.Name.String() {
+					to, err := t.Apply(diff)
+					if err != nil {
+						return err
+					}
+					toCreateTableEntity, ok := to.(*CreateTableEntity)
+					if !ok {
+						return ErrEntityTypeMismatch
+					}
+					s.tables[i] = toCreateTableEntity
+					found = true
+					break
+				}
+			}
+			if !found {
+				return ErrApplyTableNotFound
+			}
+		case *AlterViewEntityDiff:
+			// We expect the view to exist
+			found := false
+			for i, v := range s.views {
+				if v.ViewName.Name.String() == diff.from.ViewName.Name.String() {
+					to, err := v.Apply(diff)
+					if err != nil {
+						return err
+					}
+					toCreateViewEntity, ok := to.(*CreateViewEntity)
+					if !ok {
+						return ErrEntityTypeMismatch
+					}
+					s.views[i] = toCreateViewEntity
```

Comment:
Same here, does `s.named` need to be updated?

Reply:
same as above, there is no need.